lib/fs.js

```diff
@@ -513,13 +513,12 @@ function defaultCloseCallback(err) {
  * @returns {void}
  */
 function close(fd, callback = defaultCloseCallback) {
-  fd = getValidatedFd(fd);
   if (callback !== defaultCloseCallback)
     callback = makeCallback(callback);
 
   const req = new FSReqCallback();
   req.oncomplete = callback;
-  binding.close(fd, req);
+  binding.close(getValidatedFd(fd), req);
```

Comment:
Previously any errors thrown by `getValidatedFd(...)` would have thrown before creating the `FSReqCallback()`, which  I think is desirable here.

Reply:
Yes, that's the intended change, which helped finding faulty tests that depend on code execution order to validate inputs. I think the question we should ask ourselves is: why do we initialize a class just to pass a variable

Reply:
The class is there to handle proper async callback rituals (e.g. run process ticks and all). There's a lot more to just invoking the sys call when it comes to async operations like this one.